x/concentrated-liquidity/position.go

```diff
@@ -118,26 +140,60 @@ func (k Keeper) setPosition(ctx sdk.Context,
 	positionId uint64,
 ) {
 	store := ctx.KVStore(k.storeKey)
-	key := types.KeyFullPosition(poolId, owner, lowerTick, upperTick, joinTime, freezeDuration, positionId)
-	osmoutils.MustSetDec(store, key, liquidity)
+
+	// Create a new Position object with the provided information.
+	position := model.Position{
+		PositionId:     positionId,
+		PoolId:         poolId,
+		Address:        owner.String(),
+		LowerTick:      lowerTick,
+		UpperTick:      upperTick,
+		JoinTime:       joinTime,
+		FreezeDuration: freezeDuration,
+		Liquidity:      liquidity,
+	}
+
+	// Set the position ID to position mapping.
+	key := types.KeyPositionId(positionId)
+	osmoutils.MustSet(store, key, &position)
+
+	// Set the address-pool-position ID to position mapping.
+	key = types.KeyAddressPoolIdPositionId(owner, poolId, positionId)
+	store.Set(key, sdk.Uint64ToBigEndian(positionId))
+
+	// Set the pool ID to position ID mapping.
+	key = types.KeyPoolPositionPositionId(poolId, positionId)
+	store.Set(key, sdk.Uint64ToBigEndian(positionId))
```

Comment:
Note to reviewer: this data field is not currently used / tested but was desired in the issue / spec. I suggest when queries are made that utilize this data is when the respective tests are made for it, rather than in this PR.

Reply:
Let's track this in issue please

Reply:
Issue added! https://github.com/osmosis-labs/osmosis/issues/4719